src/scanner.c

```diff
@@ -359,11 +359,11 @@ static Char GetCleanedChar( UInt *wasEscaped ) {
     if      ( *STATE(In) == '\n')
       return GetCleanedChar(wasEscaped);
     else if ( *STATE(In) == '\r' )  {
-      GET_CHAR();
-      if  ( *STATE(In) == '\n' )
+      if ( PEEK_CHAR() == '\n' ) {
+        GET_CHAR();
         return GetCleanedChar(wasEscaped);
+      }
       else {
-        UNGET_CHAR(*STATE(In));
```

Comment:
This `UNGET_CHAR` call really should be `UNGET_CHAR('\r');`. But at least on Unix, we very rarely see `\r` to start with, so it's unlikely to be triggered. Of course this code is meant to deal with CRLF line endings used on Windows. Problems will only arise if there is a lone `\r`, as was common in classic MacOS files (which used CR line ends), but since that OS has died out a decade ago or so, I guess few people will hit it.

But of course we can craft a file or string containing it. But then what happens is that we return `\r` below, and I think in all cases that already leads to an error, so hopefully we are fine?

Better to fix this using `PEEK_CHAR`, though, and not have to worry.